test/comps/test_rf.cc

```diff
@@ -0,0 +1,163 @@
+#include "RandLAPACK.hh"
+#include "rl_blaspp.hh"
+#include "rl_lapackpp.hh"
+
+#include <RandBLAS.hh>
+
+#include <fstream>
+#include <gtest/gtest.h>
+
+class TestRF : public ::testing::Test
+{
+    protected:
+
+    virtual void SetUp() {};
+
+    virtual void TearDown() {};
+
+    template <typename T>
+    struct RFTestData {
+        std::vector<T> A;
+        std::vector<T> A_cpy;
+        std::vector<T> Q;
+        std::vector<T> Buf1;
+        std::vector<T> Buf2;
+        std::vector<T> Q_cpy;
+        std::vector<T> Q_hat_cpy;
+
+        RFTestData(int64_t m, int64_t n, int64_t k) :
+        A(m * n, 0.0), 
+        A_cpy(m * n, 0.0),
+        Q(m * k, 0.0), 
+        Buf1(m * m, 0.0), 
+        Buf2(m * m, 0.0), 
+        Q_cpy(m * k, 0.0), 
+        Q_hat_cpy(m * n, 0.0) 
+        {}
+    };
+
+    template <typename T, typename RNG>
+    static void computational_helper(int64_t m, int64_t n, RFTestData<T>& all_data) {
+        
+        lapack::lacpy(MatrixType::General, m, n, all_data.A.data(), m, all_data.A_cpy.data(), m);
+        
+        RandLAPACK::CholQRQ<T> CholQRQ(false, false);
+        CholQRQ.call(m, n, all_data.A_cpy);
+
+        lapack::lacpy(MatrixType::General, m, n, all_data.A_cpy.data(), m, all_data.Q_hat_cpy.data(), m);
+    }
+
+    /// General test for QB:
+    /// Computes QB factorzation, and checks:
+    /// 1. A - QB
+    /// 2. B - \transpose{Q}A
+    /// 3. I - \transpose{Q}Q
+    /// 4. A_k - QB = U_k\Sigma_k\transpose{V_k} - QB
+    template <typename T, typename RNG>
+    static void test_RF_general(int64_t m, int64_t n, int64_t k, int64_t p, RandBLAS::base::RNGState<RNG> state, RFTestData<T>& all_data) {
+
+        printf("|==================================TEST QB2 GENERAL BEGIN==================================|\n");
```

Comment:
This printf says ``QB2`` when it should say ``RF``.

Reply:
My bad. I'm actually getting rid of all those prints.